src/charm.py

```diff
@@ -179,6 +151,62 @@ def main(self, event):
 
         self.model.unit.status = ActiveStatus()
 
+    def _configure_mesh(self):
+        if self.interfaces["ingress"]:
+            self.interfaces["ingress"].send_data(
+                {
+                    "prefix": "/",
+                    "rewrite": "/",
+                    "service": self.model.app.name,
+                    "port": self.model.config["port"],
+                }
+            )
+
+    def _check_model_name(self):
+        if self.model.name != "kubeflow":
+            # Remove when this bug is resolved: https://github.com/kubeflow/kubeflow/issues/6136
+            self.model.unit.status = BlockedStatus(
+                "kubeflow-dashboard must be deployed to model named `kubeflow`:"
+                " https://git.io/J6d35"
+            )
+            raise CheckFailed
+
+    def _check_leader(self):
+        if not self.unit.is_leader():
+            # We can't do anything useful when not the leader, so do nothing.
+            self.model.unit.status = WaitingStatus("Waiting for leadership")
+            raise CheckFailed
+
+    def _check_interfaces(self):
+        try:
+            self.interfaces = get_interfaces(self)
```

Comment:
This is a pylint error: https://vald-phoenix.github.io/pylint-errors/plerr/errors/classes/W0201.html. Linting isn't something that needs to be blindly followed, and our class is slightly different than normal Python since the operator framework wants us to avoid doing anything in `__init__`. However, I don't see any reason not to just set this as a local variable in `main`.

Also, with https://github.com/canonical/serialized-data-interface/pull/24 being released, this could just be `ingress = self.get_interface(self, 'ingress')` in `main`, instead of being a whole separate `_check_interfaces` . Since it's already in a `try`/`except` block, you wouldn't even need to have the explicit `except NoCompatibleVersions as err:`-style checks here.

Reply:
I agree, with the `self.interface` part and changed it.

I don't really understand how we could avoid the style checks and status setting with `self.get_interface`. I would prefer to keep it this way for now, as it remains similar to our previous code and it will make applying these changes in all the other charms a lot easier.